sled-agent/src/instance.rs

```diff
@@ -585,10 +600,17 @@ impl Instance {
         // Create the propolis zone and resources
         let setup = self.setup_propolis_locked(&mut inner).await?;
 
+        let ws_uri = setup.client.instance_serial_console_ws_uri();
+
         // Ensure the instance exists in the Propolis Server before we start
         // using it.
         inner.ensure(self.clone(), ticket, setup, migrate).await?;
 
+        if inner.serial_tty_task.is_none() {
+            inner.serial_tty_task =
+                Some(SerialConsoleBuffer::new(ws_uri, inner.log.clone()));
+        }
+
```

Comment:
I'd move this into `InstanceInner::ensure` to better match the current instance setup